public/javascripts/itemDetailView.js

```diff
@@ -5,10 +5,19 @@
 var itemDetailView = {
     productID: null,
     item: null,
+    prevPage: null,
 
     init: function () {
         $("#response").text("");
-        this.productID = window.args.ProductID;
+
```

Comment:
This can be shortened to:
this.productID = window.args.ProductID || window.state.ProductID;
this.prevPage = window.args.PreviousPage || this.prevPage = window.state.PreviousPage;
if(window.args.ProductID){
navigation.saveState(window.args);
}